maths/haversine_distance.nim

```diff
@@ -0,0 +1,50 @@
+# Haversine formula
+
+import std/[math]
+
+func haversineDistance(latitudeA, longitudeA, latitudeB,
+    longitudeB: float): float =
+  ## returns the length of the shortest path connecting the input points on an unit sphere.
+  ## The input points are represented by their spherical/geographical coordinates.
+  let dLatitude = latitudeB - latitudeA
+  let dLongitude = longitudeB - longitudeA
+  let a = sin(dLatitude / 2.0)^2+cos(latitudeA)*cos(latitudeB)*sin(dLongitude / 2.0)^2
```

Comment:
Funny thing: if I:
```suggestion
  let a = sin(dLatitude /2.0)^2+cos(latitudeA)*cos(latitudeB)*sin(dLongitude / 2.0)^2
```
I get:
```
.../Nim/maths/haversine_distance.nim(11, 25) Error: type mismatch
Expression: /2.0
  [1] 2.0: float64

Expected one of (first mismatch at [position]):
[1] proc `/`(x, y: int): float
[2] proc `/`(x, y: float): float
[2] proc `/`(x, y: float32): float32
```

I am using
```
Nim Compiler Version 2.0.0 [Linux: amd64]
Compiled at 2023-08-01
Copyright (c) 2006-2023 by Andreas Rumpf

git hash: a488067a4130f029000be4550a0fb1b39e0e9e7c
active boot switches: -d:release
```

Reply:
What's funny about it?
```nim
let (a, b) = (1.0, 2.0)
echo a / b # 0.5
echo a /b  # parsed as ```echo(a, b.`/`)```
```

Reply:
You are right. It is not funny at all.

Reply:
Now we're talking! :)